extensions/mssql/package.json

```diff
@@ -418,6 +418,29 @@
     "connectionProvider": {
       "providerId": "MSSQL",
       "displayName": "%mssql.provider.displayName%",
+      "iconPath": [
+        {
+          "when": "isCloud == false && isBigDataCluster == false",
```

Comment:
2 questions:
- I thought we'd discussed putting this fallback logic into the core? Why do we need to duplicate the icons here?
- Why are the icon names different for light vs dark?

Reply:
I should have explained with more details, sorry for that.
### What is `default_server.svg` icon for?
`default_server.svg` is a final fallback default icon for a server when:
* we do not know any information about the server. Server could be PostGreSql, SQL Server, or other type.. We just do not know yet.
* Or, the server context does not match any `when` clause in provided connectionProvider icon path.

The ***default icon CSS rule*** is defined at `src\sql\workbench\parts\dataExplorer\browser\media\connectionViewletPanel.css`, and is shown to screen when a `div` element has `icon` and `server-page` class.

### How does a server icon falls back to `default_server.svg`?

If there is a matched `when` clause in connectionProvider iconPath config, `IconRenderer` *creates or re-uses* a CSS rule for the matched icon, and it puts specific `id` to a targeting `div` so that the  default icon CSS rule can be overridden by the match icon CSS.

If it is before making connection to a server, or there is no matched `when` clause in icon path for the server context, `IconRenderer` does not do anything. And this behavior automatically enables the default icon CSS rule.

### Why do we need `sql_stand_alone.svg`?
As, I mentioned earlier, `default_server.svg` is a final-fallback-miscellaneous-default icon which server could be ***any*** type. SQL Server stand alone is not a miscellaneous type, and is supposed to have specific icon for it.

For now, `default_server.svg` and `sql_stand_alone.svg` are looking the same, and we may want to make the different in the future.

### What is different between `server_page_xxx.svg` vs `default_server.svg`?
`default_server.svg` is `server_page_xxx.svg` without connection status circles.
`server_page_connected.svg` and `server_page_disconnected.svg` are icons with connection status hard-printed in the icons. Since we replaced the connection status circles to badges, we need to remove it from the icons.

Reply:
> Why are the icon names different for light vs dark?
Fixed.